lib/domains/character/CharacterFactory.tsx

```diff
@@ -78,12 +106,40 @@ export const CharacterFactory = {
       return character;
     }
   },
-  duplicate(c: ICharacter): ICharacter {
+  duplicate(character: ICharacter): ICharacter {
     return {
-      ...c,
+      ...character,
       id: Id.generate(),
+      pages: character.pages.map((page) => {
+        return {
+          ...page,
+          id: Id.generate(),
+          rows: page.rows.map((row) => {
+            return {
+              ...row,
+              columns: row.columns.map((column) => {
+                return {
+                  ...column,
+                  sections: column.sections.map((section) => {
+                    return {
+                      ...section,
+                      id: Id.generate(),
+                      pages: section.blocks.map((block) => {
```

Comment:
**must**: i think here its supposed to be `blocks`

Reply:
You're right. I missed that edit moving from local to here.